### PR TITLE
fix: Wallet Attestation response code and method

### PR DESCRIPTION
**Pull request by grausof**
This PR aligns the Wallet Attestation request with the REST specification

### CODE REVIEW

**Review by peppelinux — APPROVED** [review]

**Review by fmarino-ipzs — CHANGES_REQUESTED** [review]
Please, add the normative text according to the non normative examples

**Comment by grausof** [comment]
> Please, add the normative text according to the non normative examples

Can you give an example? Originally there was no other additional text

**Comment by peppelinux** [comment]
> > Please, add the normative text according to the non normative examples
> 
> Can you give an example? Originally there was no other additional text

During the weekly meeting we agreed that along with the non normative example we need to better clarify the beahviour of the implementations using normative language and clear definition about the requirements these must satisfy

therefore we decided that this PR requires normative language within the appropriate section describing the result represented in the non normative examples

**Review by fmarino-ipzs — APPROVED** [review]

File: docs/en/wallet-attestation.rst [hunk 1]
```diff
@@ -118,14 +118,31 @@ Wallet Instance Initialization and Registration
 
 **Step 9**: The Wallet Instance sends the ``challenge`` with Key Attestation and Cryptographic Hardware Key Tag to the Wallet Provider Backend in order to register the Wallet Instance identified with the Cryptographic Hardware Key public key.
 
-.. note::
+The request to the Wallet Provider endpoint in order to register the Wallet Instance MUST use the HTTP POST method with the parameters in the message body encoded in ``application/json`` format and the following parameters MUST be provided:
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
In order to register the Wallet Instance, the request to the Wallet Provider MUST use the HTTP POST method. The parameters MUST be encoded using the `application/json` format and included in the message body. The following parameters MUST be provided:
```

I'd propose to give a name to the wallet registration endpoint in the wallet solution architecture

File: docs/en/wallet-attestation.rst [hunk 2]
```diff
@@ -118,14 +118,31 @@ Wallet Instance Initialization and Registration
...
+      - **Description**
+      - **Reference**
+    * - **challenge**
+      - MUST be set to the challenge obtained from the Wallet Provider throught the ``nonce`` endpoint.
+      - `OAuth 2.0 Nonce Endpoint`_
+    * - **key_attestation**
+      - It MUST be a ``base64url`` encoded Key Attestation obtained from the **Device Integrity Service**.
+      -
+    * - **hardware_key_tag**
+      - It MUST be set to the unique identifier of the **Cryptographic Hardware Keys** in ``base64url`` encoded
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
      - It MUST be set with the unique identifier of the **Cryptographic Hardware Keys** and encoded in ``base64url``.
```

File: docs/en/wallet-attestation.rst [hunk 3]
```diff
@@ -152,10 +169,12 @@ Wallet Instance Initialization and Registration
   4. If these checks are passed, it MUST register the Wallet Instance, keeping the Cryptographic Hardware Key Tag and all useful information related to the device.
   5. It SHOULD associate the Wallet Instance with a specific User uniquely identified within the Wallet Provider's systems. This will be useful for the lifecycle of the Wallet Instance and for a future revocation.
 
+The response, in case of success, from the Wallet Provider must necessarily be a 204 No content.
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
Upon successful registration of the Wallet Instance, the Wallet Provider MUST respond with a status code set to 204 (No Content).
```

File: docs/en/wallet-attestation.rst [hunk 4]
```diff
@@ -352,10 +371,15 @@ Below an non-normative example of the Wallet Attestation without encoding and si
   }
 
 **Step 18**: The Wallet Instance receives the Wallet Attestation signed by the Wallet Provider and performs security and integrity verifications.
+The response is returned by the Wallet Provider at the end of the issuance flow.
+If the authentication is successful the http response code MUST be 200 OK.
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
**Step 18**: The response is returned by the Wallet Provider. If successful, the HTTP response code MUST be set to 200 OK and contain the Wallet Attestation signed by the Wallet Provider. The Wallet Instance therefore performs security and integrity verification about the Wallet Attestation. 
```

here we should be more accurate providing which security and integrity verification are supposed to be made by the wallet instance about the wallet attestation received

File: docs/en/wallet-attestation.rst [hunk 5]
```diff
@@ -352,10 +371,15 @@ Below an non-normative example of the Wallet Attestation without encoding and si
...
+
+Below is a non-normative example of the response.
 
 .. code-block:: http
 
-    HTTP/1.1 201 OK
+    HTTP/1.1 200 OK
     Content-Type: application/jwt
 
     eyJhbGciOiJFUzI1NiIsInR5cCI6IndhbGx... redacted
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
    eyJhbGciOiJFUzI1NiIsInR5cCI6IndhbGx ... 
```

**Pull request merged**